src/composables/permissions.ts

```diff
@@ -149,17 +157,28 @@ export function usePermissions() {
       protocol: url.protocol,
       url: url.href,
     };
+    const props = { ...modalProps, app };
 
     try {
-      const { modal, popup } = modalAndPopupTypes[method] || {};
+      let { modal, popup } = modalAndPopupTypes[method] || {};
       if (!modal || !popup) {
         return false;
       }
 
+      // The wallet app is able to recognize and properly display only bunch of transaction types.
+      // For every other we need to display different modal with some warnings.
+      if (
+        method === METHODS.sign
+        && (!modalProps.txBase64 || !isTxOfASupportedType(modalProps.txBase64))
+      ) {
+        modal = MODAL_CONFIRM_RAW_SIGN;
+        popup = POPUP_TYPE_RAW_SIGN;
+      }
```

Comment:
Previously the `openPopup` was detecting if the transaction is supported by the app or not. The same check was done in the previous permissions Vuex module. So there was 2 places doing the same logic. I consolidated this and moved here as the permissions composable is the only place where the `openPopup` function is fired.